src/main/java/sirius/web/util/BarcodeController.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.web.util;
+
+import com.google.zxing.BarcodeFormat;
+import com.google.zxing.Writer;
+import com.google.zxing.WriterException;
+import com.google.zxing.client.j2se.MatrixToImageWriter;
+import com.google.zxing.common.BitMatrix;
+import com.google.zxing.datamatrix.DataMatrixWriter;
+import com.google.zxing.oned.Code128Writer;
+import com.google.zxing.oned.EAN13Writer;
+import com.google.zxing.qrcode.QRCodeWriter;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import sirius.kernel.commons.Strings;
+import sirius.kernel.di.std.Register;
+import sirius.web.controller.BasicController;
+import sirius.web.controller.Routed;
+import sirius.web.http.MimeHelper;
+import sirius.web.http.WebContext;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+/**
+ * Used to generate barcodes by responding to "/qr" or "/barcode".
+ */
+@Register
+public class BarcodeController extends BasicController {
+
+    /**
+     * Creates an QR code for the given content.
+     * <p>
+     * The parameter <tt>content</tt> determines the contents of the qr code. The parameters <tt>with</tt> and
+     * <tt>height</tt> its dimensions.
+     * </p>
+     *
+     * @param ctx the current request
+     * @throws Exception in case an error occurred when generating the qr code
+     */
+    @Routed(value = "/qr", priority = 999)
+    public void qr(WebContext ctx) throws Exception {
```

Comment:
Optional: rename `ctx` to `webContext` in this file

Reply:
:+1: would also be consistent with line 53 then